comp/metadata/inventoryotel/inventoryotelimpl/inventoryotel.go

```diff
@@ -0,0 +1,256 @@
+		//       triggered by FA, so maybe this is OK.
+		//
+		// We want to be notified when the configuration is updated
+		deps.Config.OnUpdate(func(_ string, _, _ any) { i.Refresh() })
+	}
+
+	return provides{
+		Comp:                 i,
+		Provider:             i.MetadataProvider(),
+		FlareProvider:        i.FlareProvider(),
+		StatusHeaderProvider: status.NewHeaderInformationProvider(i),
+	}
+}
+
+func scrub(s string) string {
+	// Errors come from internal use of a Reader interface. Since we are reading from a buffer, no errors
+	// are possible.
+	scrubString, _ := scrubber.ScrubString(s)
+	return scrubString
+}
+
+func (i *inventoryotel) parseResponseFromJSON(body []byte) (otelMetadata, error) {
+	var conf interface{}
+
+	err := json.Unmarshal(body, &conf)
+	if err != nil {
+		i.log.Errorf("Error unmarshaling the response:", err)
+		return nil, err
+	}
+
+	return conf.(otelMetadata), nil
+}
+
+func (i *inventoryotel) fetchRemoteOtelConfig(u *url.URL) (otelMetadata, error) {
+	// Create a Bearer string by appending string access token
+	var bearer = "Bearer " + i.authToken.Get()
+
+	// Create a new request using http
+	req, err := http.NewRequest("GET", u.String(), nil)
+	if err != nil {
+		i.log.Errorf("Error building request: ", err)
+		return nil, err
+	}
+
+	// add authorization header to the req
+	req.Header.Add("Authorization", bearer)
+
+	client := &http.Client{}
+	resp, err := client.Do(req)
+	if err != nil {
+		i.log.Errorf("Error on response: ", err)
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		i.log.Errorf("Error while reading the response bytes:", err)
+		return nil, err
+	}
+
+	return i.parseResponseFromJSON(body)
+
+}
+
+func (i *inventoryotel) fetchDummyOtelConfig(_ *url.URL) (otelMetadata, error) {
+	dummy, err := dummyFS.ReadFile(path.Join("dummy_data", "response.json"))
+	if err != nil {
+		i.log.Errorf("Unable to read embedded dummy data:", err)
+		return nil, err
+	}
+
+	return i.parseResponseFromJSON(dummy)
+}
+
+func (i *inventoryotel) fetchOtelAgentMetadata() {
+	isEnabled := i.conf.GetBool("otel.enabled")
+
+	if !isEnabled {
+		i.log.Infof("OTel Metadata unavailable as OTel collector is disabled")
+		i.data = nil
+
+		return
+	}
+	data, err := i.f.getConfig()
+	if err != nil {
+		i.log.Errorf("Unable to fetch fresh inventory metadata: ", err)
+	}
+
+	i.data = data
+	if i.data == nil {
+		i.log.Infof("OTel config returned empty")
+		return
+	}
+
+	i.data["otel_enabled"] = isEnabled
+
+}
+
+func (i *inventoryotel) refreshMetadata() {
+	// Core Agent / agent
+	i.fetchOtelAgentMetadata()
+}
+
+func (i *inventoryotel) getPayload() marshaler.JSONMarshaler {
+	i.m.Lock()
+	defer i.m.Unlock()
+
+	i.refreshMetadata()
+
+	// Create a static scrubbed copy of agentMetadata for the payload
+	data := copyAndScrub(i.data)
+
+	return &Payload{
+		Hostname:  i.hostname,
+		Timestamp: time.Now().UnixNano(),
+		Metadata:  data,
+		UUID:      uuid.GetUUID(),
```

Comment:
are we storing this field in redapl ?

Reply:
These can be used by the `event-platform-resource-writer` but are not stored in REDAPL.